### DISCUSSION TITLE
Incorrect match info for arrays

### DISCUSSION THREAD

**Discussion by interlark** [meilisearch/product#121] [Feedback & Feature Proposal]
1.  Add documents
      ```json
      [  
         {
           "id": 1,
           "name": "Hello",
           "responses": ["Salut", "What's up"]
         },
         {
           "id": 2,
           "name": "Bye",
           "responses": ["Later", "Sayonara"]
         }
      ]
      ```
2. Try to search for items inside those arrays like **sayonara** or **later** and pay attention to the **_matchesInfo**. It would not give you information about a match, only useless **start** and **length** with which you can not get what exactly was found in the array.
Here's an example for ```search?q=sayonara&matches=true```:
      ```json
      {
        "hits": [
          {
            "id": 2,
            "name": "Bye",
            "responses": [
              "Later",
              "Sayonara"
            ],
            "_matchesInfo": {
              "responses": [
                {
                  "start": 0,
                  "length": 8
                }
              ]
            }
          }
        ],
        "nbHits": 1,
        "exhaustiveNbHits": false,
        "query": "sayonara",
        "limit": 20,
        "offset": 0,
        "processingTimeMs": 0
      }
      ```
So you don't know to what element in array this match info belongs thus you can not use this match info.
`{ "start": 0, "length": 8 } ` is related to `Later` or `Sayonara` ?

**Comment by curquiza** [comment]
Hello @interlark!
Sorry for this inconvenience, `matches` does not work for objects and arrays, only for strings.
This is indeed not clear at all since it's not in the docs, I'm going to add this in the docs! This will be present in the docs of the next version of MeiliSearch (v0.21.0)

`matches` on arrays is something we might want to handle in the future, I let @gmourier decide 🙂

**Comment by CaroFG** [comment]
Just for info, I tested this with v0.20 and I get the following response: 
```
{
  "hits": [
    {
      "id": 2,
      "name": "Bye",
      "responses": [
        "Later",
        "Sayonara"
      ],
      "_matchesInfo": {
        "responses": [
          {
            "start": 7,
            "length": 8
          }
        ]
      }
    }
  ],
  "offset": 0,
  "limit": 20,
  "nbHits": 1,
  "exhaustiveNbHits": false,
  "processingTimeMs": 0,
  "query": "sayonara"
}
```

Instead of getting `"start": 0`, I got `"start": 7`
